### PR TITLE
Boost 1.83 survey

### PR DESCRIPTION
**Pull request by liushuyu**
Topic Description
-----------------

Boost 1.83 survey

Package(s) Affected
-------------------

```
runtime-common/boost
app-games/0ad
app-multimedia/aegisub
app-admin/aptitude
app-devel/binutils+cross-mips64r6el
runtime-display/imath
runtime-imaging/openexr
runtime-common/openvdb
runtime-imaging/openimageio
app-devel/cuda
app-creativity/blender
app-games/citra
app-devel/cpprestsdk
runtime-i18n/libime
app-i18n/fcitx5-chinese-addons
runtime-imaging/pcl
app-creativity/freecad
runtime-editors/source-highlight
app-devel/gdb+cross-arm64
app-devel/gdb+cross-mips64r6el
app-utils/gpick
app-devel/heaptrack
app-creativity/hugin
app-web/i2pd
app-creativity/inkscape
app-utils/innoextract
runtime-scientific/ixion
runtime-productivity/libkolabxml
desktop-kde/kdepim-runtime
desktop-kde/kig
runtime-desktop/libappimage
runtime-productivity/libcmis
runtime-productivity/liborcus
runtime-i18n/libphonenumber
runtime-productivity/libpst
runtime-i18n/librime
runtime-web/libtorrent-rasterbar
runtime-ham/libuhd
runtime-common/lucene++
app-multimedia/mkvtoolnix
app-games/mupen64plus
app-multimedia/ncmpcpp
app-electronics/prjtrellis
app-electronics/nextpnr
runtime-web/nghttp2
runtime-scientific/openbabel
app-games/openmw
extra-scientific/cgal
app-creativity/openscad
app-multimedia/pamixer
app-i18n/poedit
app-imaging/povray
runtime-common/process-cpp
app-electronics/pulseview
lang-python/pycuda
lang-python/pygattlib
app-games/rlvm
app-admin/snapper
app-multimedia/sunshine
app-creativity/superslicer
app-games/supertux
desktop-trinity/tdeedu
app-admin/thin-provisioning-tools
app-admin/thunderbolt-software-user-space
app-network/trojan
runtime-scientific/vigra
app-games/wesnoth
app-games/yuzu
app-multimedia/zxtune
```

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by jiegec** [comment]
Please apply https://github.com/boostorg/context/commit/819c2d6423b4e0f55c5f69e334bf81570e82f68f for loongarch64/riscv64 platforms, otherwise Boost.Context won't work.

**Comment by liushuyu** [comment]
> Please apply [boostorg/context@819c2d6](https://github.com/boostorg/context/commit/819c2d6423b4e0f55c5f69e334bf81570e82f68f) for loongarch64/riscv64 platforms, otherwise Boost.Context won't work.

Done in [c2c8c25](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4759/commits/c2c8c25c578f3eb675d853d9ebe787bfccd4cba3)

File: app-creativity/blender/autobuild/patches/HACK-impersonate-nvcc-gcc-13.patch.deferred
Comment (MingcongBai) [inline app-creativity/blender/autobuild/patches/HACK-impersonate-nvcc-gcc-13.patch.deferred:1]: Fix up the file name here.

File: app-electronics/apicula/autobuild/prepare [hunk 1]
```diff
@@ -1,5 +1,8 @@
 abinfo "Extracting prebuilt chip databases ..."
-for i in apycula/GW1N-1.pickle apycula/GW1N-4.pickle apycula/GW1N-9.pickle apycula/GW1NS-2.pickle
+DEVICES=(apycula/GW1N-1.pickle apycula/GW1N-9.pickle apycula/GW1N-4.pickle 
+	 apycula/GW1NS-2.pickle apycula/GW1NS-4.pickle apycula/GW1N-9C.pickle 
+	 apycula/GW1NZ-1.pickle apycula/GW2A-18.pickle apycula/GW2A-18C.pickle)
+for i in "${DEVICES[@]}"
 do
 	unzip ../apycula.whl $i
```
Comment (MingcongBai) [inline app-electronics/apicula/autobuild/prepare:7]: Absolute path here.

File: app-creativity/superslicer/autobuild/patches/superslicer-backport-to-tbb-2020.patch
Comment (MingcongBai) [inline app-creativity/superslicer/autobuild/patches/superslicer-backport-to-tbb-2020.patch:1]: Fix up file name here.

File: app-electronics/nextpnr/spec [hunk 2]
```diff
@@ -1,7 +1,7 @@
-VER=0+git20211107
-REL=4
-SRCS="git::commit=b7207b088539e7b5f34ebbd9f53f9fa14bef6e62;rename=nextpnr::https://github.com/YosysHQ/nextpnr.git \
-      git::commit=7c8ba15beb174d8def021bba6e1db06426e318a9;rename=mistral::https://github.com/Ravenslofty/mistral"
+VER=0.6
+# NOTE: nextpnr requires a specific version of mistral
```
Comment (MingcongBai) [inline app-electronics/nextpnr/spec:2]: Please add a comment on how to track mistral versions. Also, consider adding mistral version to the main VER.

File: app-electronics/prjoxide/autobuild/defines [hunk 3]
```diff
@@ -5,4 +5,5 @@ BUILDDEP="llvm rustc"
 PKGDES="Libraries and utilities for open-source toolchain of Lattice Nexus series FPGAs"
 
 USECLANG=1
+NOLTO__LOONGARCH64=1
```
Comment (MingcongBai) [inline app-electronics/prjoxide/autobuild/defines:9]: Add a FIXME comment here.

File: app-games/0ad/autobuild/build [hunk 4]
```diff
@@ -1,6 +1,7 @@
 (
     abinfo "Configuring $PKGNAME ..."
     cd "$SRCDIR"/build/workspaces
+    unset RUSTFLAGS
```
Comment (MingcongBai) [inline app-games/0ad/autobuild/build:5]: Please add a FIXME comment here explaining why.

File: app-games/mupen64plus/autobuild/build [hunk 5]
```diff
@@ -12,6 +12,11 @@ if [[ "${CROSS:-$ARCH}" = "armv7hf" || \
     NEON="NEON=1 VFP_HARD=1"
 fi
 
+if [[ "${CROSS:-$ARCH}" = "ppc64el" ]]; then
```
Comment (MingcongBai) [inline app-games/mupen64plus/autobuild/build]: Please use `ab_match_arch`.

File: app-multimedia/zxtune/autobuild/build [hunk 6]
```diff
@@ -1,30 +1,24 @@
...
-    release=1 \
-    pic=1 \
-    arch='' \
-    distro='' \
-    prebuilt.dir=/usr/lib/ \
-    qt.includes='/usr/include/qt5 /usr/include/qt5/QtWidgets/ /usr/include/qt5/QtGui/' \
-    CXX_MODE_FLAGS="${CPPFLAGS} ${CXXFLAGS} -fPIC -DNDEBUG" \
-    LD_MODE_FLAGS="${LDFLAGS}" \
-    -C apps/bundle/
+make "${OPTIONS[@]}" -C apps/bundle/
```
Comment (MingcongBai) [inline app-multimedia/zxtune/autobuild/build]: Absolute path here.

File: app-multimedia/zxtune/autobuild/build [hunk 7]
```diff
@@ -1,30 +1,24 @@
...
-    arch='' \
-    distro='' \
-    prebuilt.dir=/usr/lib/ \
-    qt.includes='/usr/include/qt5 /usr/include/qt5/QtWidgets/ /usr/include/qt5/QtGui/' \
-    CXX_MODE_FLAGS="${CPPFLAGS} ${CXXFLAGS} -fPIC -DNDEBUG" \
-    LD_MODE_FLAGS="${LDFLAGS}" \
-    -C apps/bundle/
+     "${OPTIONS[@]}" \
+     DESTDIR="$PKGDIR" \
+     -C apps/bundle/
```
Comment (MingcongBai) [inline app-multimedia/zxtune/autobuild/build]: Ditto.

File: desktop-gnome/gtk-3/autobuild/defines
Comment (MingcongBai) [inline desktop-gnome/gtk-3/autobuild/defines:1]: Rebase against `stable`. This change was cherry-picked out.

File: groups/boost-rebuild-minimal
Comment (MingcongBai) [inline groups/boost-rebuild-minimal:1]: Please add a comment explaining what this is used for.

File: runtime-common/openvdb/autobuild/defines [hunk 8]
```diff
@@ -6,12 +6,14 @@ BUILDDEP="doxygen llvm texlive"
 
 #FIXME: OpenVDB AX complains about too new LLVM (15+)
 #       Setting -DUSE_AX=OFF for now
+#FIXME: Due to tbb CMake configurations problem
```
Comment (MingcongBai) [inline runtime-common/openvdb/autobuild/defines:9]: Can we fix tbb?

File: runtime-common/physfs/autobuild/beyond [hunk 9]
```diff
@@ -1,6 +1,9 @@
-install -d "$PKGDIR"/usr/share/{doc/physfs,man/man3}
-install -m644 docs/html/* "$PKGDIR"/usr/share/doc/physfs
-install -m644 docs/man/man3/* "$PKGDIR"/usr/share/man/man3
+abinfo "Generating documentation ..."
+cmake --build "$SRCDIR" -- docs
```
Comment (MingcongBai) [inline runtime-common/physfs/autobuild/beyond:2]: Absolute path here.

File: runtime-common/physfs/autobuild/beyond [hunk 10]
```diff
@@ -1,6 +1,9 @@
...
+
+install -dv "$PKGDIR"/usr/share/{doc/physfs,man/man3}
+install -Dvm644 docs/html/* "$PKGDIR"/usr/share/doc/physfs
+install -Dvm644 docs/man/man3/* "$PKGDIR"/usr/share/man/man3
```
Comment (MingcongBai) [inline runtime-common/physfs/autobuild/beyond]: Ditto.

File: runtime-display/nvidia/spec
Comment (MingcongBai) [inline runtime-display/nvidia/spec:1]: Rebase against stable, as @cthbleachbit updated NVIDIA recently for Kernel 6.6

File: runtime-i18n/librime/spec [hunk 11]
```diff
@@ -1,15 +1,14 @@
-VER=1.7.3
-REL=7
-OCTAGRAMCOMMIT=78dab3a776ee758236b702f535c78f52338fff8d
-LUACOMMIT=8b37d5541b0341a07ed7517c78df1a09268001fd
-CHARCODECOMMIT=b569184772b12965e3ebe1dfd431026951fed81c
+VER=1.9.0
+OCTAGRAMCOMMIT=a6ced5a1d623afed17284fbaa395dd3a8b019a14
```
Comment (MingcongBai) [inline runtime-i18n/librime/spec:3]: Please add a comment explaining how we should track the commits.

Comment (MingcongBai) [reply runtime-i18n/librime/spec:3]: Also, I wonder why they were bundled in the first place...

File: runtime-multimedia/libva-nvidia-driver/spec
Comment (MingcongBai) [inline runtime-multimedia/libva-nvidia-driver/spec:1]: Please rebase against stable, @RedL0tus updated this package in the Graphical Stack Survey (20231031).

File: runtime-scientific/openbabel/autobuild/beyond [hunk 12]
```diff
@@ -0,0 +1,3 @@
+abinfo "Moving CMake files to the correct location ..."
+cp -rv "$PKGDIR"/lib "$PKGDIR"/usr/
+rm -r "$PKGDIR"/lib
```
Comment (MingcongBai) [inline runtime-scientific/openbabel/autobuild/beyond]: Verbosity here.

File: runtime-scientific/openbabel/autobuild/beyond
Comment (MingcongBai) [inline runtime-scientific/openbabel/autobuild/beyond:1]: Please see if this is still needed with our recent `bash-startup` update, which now favours /usr/bin over /bin.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: runtime-common/physfs/autobuild/beyond (hunk 9 above)
Comment (liushuyu) [reply runtime-common/physfs/autobuild/beyond:2]: `docs` is a target name, not a path.

File: app-creativity/blender/autobuild/patches/HACK-impersonate-nvcc-gcc-13.patch.deferred
Comment (liushuyu) [reply app-creativity/blender/autobuild/patches/HACK-impersonate-nvcc-gcc-13.patch.deferred:1]: This is done on purpose. This patch needs to be applied out-of-sequence.

File: app-electronics/nextpnr/spec (hunk 2 above)
Comment (liushuyu) [reply app-electronics/nextpnr/spec:2]: > Please add a comment on how to track mistral versions. Also, consider adding mistral version to the main VER.

I don't know how to track mistral versions. This is purely trial and error.

File: runtime-common/openvdb/autobuild/defines (hunk 8 above)
Comment (liushuyu) [reply runtime-common/openvdb/autobuild/defines:9]: Not in this topic. It would require a separate rebuild

**Review by MingcongBai — DISMISSED** [review]

**Review by MingcongBai — APPROVED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**